compatibility_spark-3/src/test/scala/za/co/absa/hyperdrive/compatibility/impl/writer/cdc/CDCTestBase.scala

```diff
@@ -13,16 +13,28 @@
  * limitations under the License.
  */
 
-package za.co.absa.hyperdrive.compatibility.impl.writer.delta
+package za.co.absa.hyperdrive.compatibility.impl.writer.cdc
 
+import org.apache.spark.sql.SparkSession
 import org.apache.spark.sql.execution.streaming.MemoryStream
 import org.scalatest.{BeforeAndAfterEach, FlatSpec}
 import za.co.absa.commons.io.TempDirectory
-import za.co.absa.hyperdrive.shared.utils.SparkTestBase
 
-trait DeltaTestBase extends BeforeAndAfterEach with SparkTestBase {
+trait CDCTestBase extends BeforeAndAfterEach {
   this: FlatSpec =>
 
+  implicit val spark: SparkSession = SparkSession.builder()
```

Comment:
Now we have duplication with `SparkTestBase`. This might be a good moment to introduce the spark-commons `SparkTestBase`. https://github.com/AbsaOSS/spark-commons/blob/master/spark-commons-test/src/main/scala/za/co/absa/spark/commons/test/SparkTestBase.scala

It should be easier to define the custom spark session without duplication

Reply:
Indeed, I see the problem now. I thought you could reuse `DefaultSparkConfiguration`, but you can't.